wpimath/src/main/native/cpp/trajectory/Trajectory.cpp

```diff
@@ -124,6 +124,21 @@ Trajectory Trajectory::RelativeTo(const Pose2d& pose) {
   return Trajectory(newStates);
 }
 
+Trajectory Trajectory::operator+(const Trajectory& other) const {
+  auto states = m_states;
+  auto otherStates = other.States();
+  for (auto& otherState : otherStates) {
+    otherState.t += m_totalTime;
+  }
+
+  // Here we omit the first state of the other trajectory because we don't want
+  // two time points with different states. Sample() will automatically
+  // interpolate between the end of this trajectory and the second state of the
+  // other trajectory.
+  states.insert(states.end(), otherStates.begin() + 1, otherStates.end());
```

Comment:
This incorrectly omits the first state of the RHS trajectory if the LHS is a default-constructed Trajectory that contains no states. This caused a weird acceleration event at the start of auton driving in 3512's tests. The following fixed it.
```suggestion
  if (states.size() > 0) {
    states.insert(states.end(), otherStates.begin() + 1, otherStates.end());
  } else {
    states.insert(states.end(), otherStates.begin(), otherStates.end());
  }
```